perfkitbenchmarker/benchmarks/fio_benchmark.py

```diff
@@ -260,51 +266,124 @@ def GenerateJobFileString(disk, against_device,
-    disk: the disk.BaseDisk object we're benchmarking with.
-    against_device: bool. True if we're using a raw disk.
-    scenario_strings: list of strings or None. The workload scenarios to
-      generate.
-    io_depths: iterable. The IO queue depths to test.
-    working_set_size: int or None. If int, the size of the working set in GB.
+    job_file_path: string or None. The path to the user's job file, if
+      provided.
+    scenario_strings: list of strings or None. The workload scenarios
+      to generate.
+    against_device: bool. True if testing against a raw device, False
+      if testing against a filesystem.
+    disk: the disk.BaseDisk object to test against.
+    io_depths: string. The IO queue depths to test.
+    working_set_size: int or None. If int, the size of the working set
+      in GB.
 
   Returns:
     A string containing a fio job file.
   """
 
-  # The default behavior is to run with the standard fio job file.
-  if scenario_strings is None and not fio_jobfile:
-    fio_jobfile = data.ResourcePath('fio.job')
+  use_user_jobfile = job_file_path or not scenario_strings
 
-  if fio_jobfile:
-    with open(fio_jobfile, 'r') as jobfile:
-      return jobfile.read()
+  if use_user_jobfile:
+    remove_filename = against_device
+    return ProcessedJobFileString(job_file_path or data.ResourcePath('fio.job'),
+                                  remove_filename)
   else:
-    if 'all' in scenario_strings:
-      scenarios = SCENARIOS.itervalues()
+    if against_device:
+      filename = disk.GetDevicePath()
     else:
-      for name in scenario_strings:
-        if name not in SCENARIOS:
-          logging.error('Unknown scenario name %s', name)
-      scenarios = (SCENARIOS[name] for name in scenario_strings)
+      # Since we pass --directory to fio, we must use relative file
+      # paths or get an error.
+      filename = DEFAULT_TEMP_FILE_NAME
 
-    return GenerateJobFileString(disk, against_device, scenarios,
+    return GenerateJobFileString(filename, scenario_strings,
                                  io_depths, working_set_size)
 
 
+def WarnOnBadFlags():
+  """Warn the user if they pass bad flag combinations."""
+
+  if FLAGS.fio_jobfile:
+    ignored_flags = {'--' + flag_name
+                     for flag_name in FLAGS_IGNORED_FOR_CUSTOM_JOBFILE
+                     if FLAGS[flag_name].present}
+
+    if ignored_flags:
+      logging.warning('Fio job file specified. Ignoring options "%s"',
+                      ', '.join(ignored_flags))
+
+  if FLAGS.run_for_minutes % MINUTES_PER_JOB != 0:
+    logging.warning('Runtime %s will be rounded up to the next multiple of %s '
+                    'minutes.', FLAGS.run_for_minutes, MINUTES_PER_JOB)
+
+  if (FLAGS.fio_jobfile is None and
+      FLAGS.generate_scenarios and
+      not FLAGS.working_set_size):
+    logging.error('You must specify the working set size when using generated '
+                  'scenarios against a filesystem.')
+    sys.exit(1)
```

Comment:
We should not exit within the benchmark. Instead we should raise an exception and let the upper level to decide.